### PR TITLE
AWS Fargate provider

### LINKED ISSUE #12: [AWS] implement provider for ECS and Fargate
### Summary
Add a provider for AWS ECS and Fargate 

### Description
Provide the ability to use kubectl to deploy a pod spec to AWS ECS and also to AWS Fargate. 

It should be possible to support deploying to both native ECS and to Fargate using the same provider. There are only slight differences is the task definitions between the two.

### PR DESCRIPTION
**Pull request by ofiliz**
This PR implements the official [AWS Fargate](https://aws.amazon.com/fargate) provider for virtual-kubelet.

This initial release includes:
* [x] Basic Kubernetes pod lifecycle on [AWS Fargate](https://aws.amazon.com/fargate)
* [x] Container name, image, command, arguments, working directory and environment variables
* [x] Container resource (CPU and memory) requests, limits and automatic mapping to Fargate task sizes
* [x] Configurable Fargate platform version and virtual node resource capacity
* [x] Image pulls from [Amazon ECR](https://aws.amazon.com/ecr) and other public registries
* [x] VPC networking with multiple subnets and a dedicated ENI for each pod
* [x] Private IP address for pods from Kubernetes pod CIDR
* [x] Optional public IP address for pods for direct internet connectivity
* [x] Security groups integration for network policy
* [x] CloudWatch Logs integration for container logs
* [x] [AWS IAM](https://aws.amazon.com/iam) roles for pods via annotations

**Important note:**
*Virtual-kubelet and the AWS Fargate virtual-kubelet provider are in very early stages of development.
DO NOT run the AWS Fargate provider in any Kubernetes or Fargate production environments.*

We would like to especially thank @johanneswuerbach for his contributions.

Fixes #12

### CODE REVIEW

File: providers/aws/config.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+// loadConfigFile loads the given Fargate provider configuration file.
+func (p *FargateProvider) loadConfigFile(filePath string) error {
+	f, err := os.Open(filePath)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	err = p.loadConfig(f)
+	return err
```
Comment (jessfraz) [inline providers/aws/config.go]: nit: could just be `return p.loadConfig(f)`

File: vkubelet/vkubelet.go [hunk 2]
```diff
@@ -38,10 +39,8 @@ type Server struct {
 
 // New creates a new virtual-kubelet server.
 func New(nodeName, operatingSystem, namespace, kubeConfig, taint, provider, providerConfig string) (*Server, error) {
-	var (
-		config *rest.Config
-		err    error
-	)
+	var config *rest.Config
```
Comment (ofiliz) [inline vkubelet/vkubelet.go]: Fixing a few golint warnings. This definition of err was not used. Also restricted a few really long lines to 120 columns.

**Comment by ofiliz** [comment]
Hello, do you think we can complete the review and merge today? This PR is touching only 2 source files outside of AWS Fargate provider, and the changes there are trivial.

As discussed in our meeting earlier this week, I am happy to be the maintainer of AWS Fargate provider and review any PRs from the community going forward.

**Comment by robbiezhang** [comment]
@ofiliz I'm good with merging it since it pass the build and basic unit tests. I plan to "Rebase and merge" the PR, so that the commit history is kept. Does it sound good to you?

**Comment by jessfraz** [comment]
lgtm, thanks!

**Comment by rbitia** [comment]
Alright I'm going to do a quick pass through and then merge.

**Pull request merged**